common/components/common/projects/EditProjectForm.jsx

```diff
@@ -0,0 +1,188 @@
+// @flow
+
+import React from 'react';
+import type {LinkInfo} from '../../../components/forms/LinkInfo.jsx'
+import type {FileInfo} from '../../common/FileInfo.jsx'
+import ImageUploadFormElement from '../../../components/forms/ImageUploadFormElement.jsx'
+import LinkList from '../../../components/forms/LinkList.jsx'
+import FileUploadList from '../../../components/forms/FileUploadList.jsx'
+import TagCategory from '../../common/tags/TagCategory.jsx'
+import TagSelect from '../../common/tags/TagSelect.jsx'
+import DjangoCSRFToken from 'django-react-csrftoken'
+import ProjectAPIUtils from '../../../components/utils/ProjectAPIUtils.js';
+import type {ProjectDetailsAPIData} from '../../../components/utils/ProjectAPIUtils.js';
+import _ from 'lodash'
+
+type FormFields = {|
+  project_name: ?string,
+  project_location: ?string,
+  project_url: ?string,
+  project_description: ?string,
+  project_links: Array<LinkInfo>,
+  project_files: Array<FileInfo>,
+  project_thumbnail: FileInfo
+|};
+
+type Props = {|
+  projectId: number
```

Comment:
it might be better to handle these as `string`s, because if we deviate from allocating IDs sequentially and just choose them randomly from our possible ID-space, the numbers might be quite large (depending on how django is configured) in which case JS will round the numbers and our IDs will be not useful

Reply:
If we do change to an id scheme that would run afoul of JS rounding, I imagine we'd probably have to come back and edit this part anyways.